packages/scxa-marker-gene-heatmap/src/MarkerGeneHeatmap.js

```diff
@@ -59,60 +59,56 @@ Highcharts.SVGRenderer.prototype.symbols.download = (x, y, w, h) => [
   const { host, chartHeight, hasDynamicHeight, heatmapRowHeight, heatmapType, species } = props
   const { cellType, data, xAxisCategories, yAxisCategories } = props
-  const totalNumberOfRows = Object.keys(_.groupBy(data, `geneName`)).length
-  const groupedData = _.groupBy(data, `cellGroupValueWhereMarker`)
 
-  const filteredData = data.map(cell => ({
+  const heatmapData = data.map(cell => ({
     ...cell,
     value: cell.value <= 0 ? null : cell.value
   }))
 
   const plotLines = []
-  const cellTypes = Object.keys(groupedData)
+  const cellTypesOrClusterIds = Object.keys(_.groupBy(data, `cellGroupValueWhereMarker`))
 
+  const totalNumberOfRows = Object.keys(_.groupBy(data, `geneName`)).length
   // 175px = title + legend + X axis labels; 8 is the height of a plot line separating the clusters
-  const dynamicHeight = (totalNumberOfRows * heatmapRowHeight) + (cellTypes.length * 8) + 175
+  const dynamicHeight = (totalNumberOfRows * heatmapRowHeight) + (xAxisCategories.length * 8) + 175
 
+  const groupedData = _.groupBy(data, `cellGroupValueWhereMarker`)
   let plotLineAxisPosition = -0.5
 
-  // If we don't have a set row height, we try to estimate the height as worked out by Highcharts
-  const rowHeight = hasDynamicHeight ?
-    heatmapRowHeight :
-    Math.round((chartHeight - 175) / totalNumberOfRows + ((cellTypes.length - 1) * 8))
-
-  cellTypes.forEach((cellType, idx, array) => {
-    const numberOfRows = Object.keys(_.groupBy(groupedData[cellType], `y`)).length // how many marker genes per cluster
-    plotLineAxisPosition = plotLineAxisPosition + numberOfRows
-
-    const yOffset = -numberOfRows * rowHeight / 2
-
-    let color, zIndex
-    // don't show last plot line
-    if (idx === array.length - 1) {
-      // removing the plot line altogether would remove the label, so we need to make it "invisible"
-      color = `#FFFFFF`
-      zIndex = 0
-    } else {
-      color = `#000000`
-      zIndex = 5
+  cellTypesOrClusterIds.length === 1 ? plotLines.push({
+    color: `#000000`,
+    width: 2,
+    value: data[0].y,
+    zIndex: 0,
+    label: heatmapType !== `multiexperimentcelltypes` && {
+      text: splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellTypesOrClusterIds[0])).join(`<br/>`),
+      align: `right`,
+      textAlign: `left`,
+      x: 0,
+      y: 30
     }
-
-    const splitCellTypeLabel = splitPhrase(heatmapOptionsProvider[heatmapType].labelsFormatter(cellType))
-    plotLines.push({
-      color: color,
-      width: 2,
-      value: plotLineAxisPosition,
-      zIndex: zIndex,
-      label: heatmapType !== `multiexperimentcelltypes` && {
-        useHTML: true,
-        text: splitCellTypeLabel.join(`<br/>`),
-        align: `right`,
-        textAlign: `left`,
-        x: 15,
-        y: yOffset - Math.max(splitCellTypeLabel.length - 3, 0) * rowHeight / 2 // Move very long labels slightly up
-      }
+  }) :
+    cellTypesOrClusterIds.map((cellTypeOrClusterId) => {
```

Comment:
`map` here isn’t returning any value, so either a different mathod or the function arugment should be changed.